Content.Shared/Lathe/SharedLatheSystem.cs

```diff
@@ -53,4 +58,29 @@ public static int AdjustMaterial(int original, bool reduce, float multiplier)
         => reduce ? (int) MathF.Ceiling(original * multiplier) : original;
 
     protected abstract bool HasRecipe(EntityUid uid, LatheRecipePrototype recipe, LatheComponent component);
+
+    private void OnPrototypesReloaded(PrototypesReloadedEventArgs obj)
+    {
+        if (!obj.WasModified<LatheRecipePrototype>())
+            return;
+        BuildInverseRecipeDictionary();
+    }
+
+    private void BuildInverseRecipeDictionary()
+    {
+        _inverseRecipeDictionary.Clear();
+        foreach (var latheRecipe in _proto.EnumeratePrototypes<LatheRecipePrototype>())
+        {
+            _inverseRecipeDictionary.TryAdd(latheRecipe.Result, new List<LatheRecipePrototype>());
+            _inverseRecipeDictionary[latheRecipe.Result].Add(latheRecipe);
+        }
```

Comment:
GetOrNew might be better as I don't know if the JIT is smart enough to not construct the list every time.